src/cairoWriter/writers/cairoContractWriter.ts

```diff
@@ -128,13 +128,11 @@ export class CairoContractWriter extends CairoASTNodeWriter {
         !(f instanceof CairoImportFunctionDefinition),
     );
 
-    const writtenImportFuncs = getGroupedImports(
-      importFunctions
-        .sort((funcA, funcB) =>
-          `${funcA.path}.${funcA.name}`.localeCompare(`${funcB.path}.${funcB.name}`),
-        )
-        .filter((func, index, importFuncs) => func.name !== importFuncs[index - 1]?.name),
-    ).reduce((writtenImports, importFunc) => `${writtenImports}\n${importFunc}`, '');
+    const writtenImportFuncs = importFunctions
+      .map((n) => writer.write(n))
```

Comment:
`.map(writer.write, writer)` would be more idiomatic

Reply:
Besides, you get incompatible types doing it that way. I don't think it's worth it.